mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/EntityHelperGetMethodGenerator.java

```diff
@@ -18,37 +18,98 @@
 import com.datastax.oss.driver.internal.mapper.processor.PartialClassGenerator;
 import com.datastax.oss.driver.internal.mapper.processor.ProcessorContext;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableMap;
 import com.squareup.javapoet.ClassName;
 import com.squareup.javapoet.MethodSpec;
 import com.squareup.javapoet.ParameterSpec;
+import com.squareup.javapoet.TypeName;
 import com.squareup.javapoet.TypeSpec;
+import java.util.Map;
 import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
 
 public class EntityHelperGetMethodGenerator implements PartialClassGenerator {
 
   private final EntityDefinition entityDefinition;
 
+  private static final Map<TypeName, String> PRIMITIVE_ACCESSORS =
+      ImmutableMap.<TypeName, String>builder()
+          .put(TypeName.BOOLEAN, "Boolean")
+          .put(TypeName.BYTE, "Byte")
+          .put(TypeName.DOUBLE, "Double")
+          .put(TypeName.FLOAT, "Float")
+          .put(TypeName.INT, "Int")
+          .put(TypeName.LONG, "Long")
+          .build();
+
+  private final EntityHelperGenerator enclosingClass;
+
   public EntityHelperGetMethodGenerator(
       EntityDefinition entityDefinition,
       EntityHelperGenerator enclosingClass,
       ProcessorContext context) {
     this.entityDefinition = entityDefinition;
+    this.enclosingClass = enclosingClass;
   }
 
   @Override
   public void addMembers(TypeSpec.Builder classBuilder) {
 
-    MethodSpec.Builder extractBuilder =
+    MethodSpec.Builder getBuilder =
         MethodSpec.methodBuilder("get")
             .addAnnotation(Override.class)
             .addModifiers(Modifier.PUBLIC)
             .addParameter(
                 ParameterSpec.builder(ClassName.get(GettableByName.class), "source").build())
-            .returns(entityDefinition.getClassName())
-            // TODO generate proper implementation (JAVA-2112)
-            .addStatement("throw new $T($S)", UnsupportedOperationException.class, "TODO");
+            .returns(entityDefinition.getClassName());
+
+    TypeName returnType = entityDefinition.getClassName();
+    String returnName = "returnValue";
+    getBuilder.addStatement("$T $L = new $T()", returnType, returnName, returnType);
+
+    int udtIndex = 0;
+    for (PropertyDefinition property : entityDefinition.getProperties()) {
+      TypeName type = property.getType();
+      String cqlName = property.getCqlName();
+      String setterName = property.getSetterName();
+      TypeElement childEntityElement = property.getEntityElement();
+      getBuilder.addCode("\n");
+      if (childEntityElement != null) {
+        udtIndex += 1;
+        // Populate udtInformation
+        String udtValueName = "udtValue" + udtIndex;
+        String valueName = "value" + udtIndex;
+        // Extract UdtValue to pass it on to underlying helper method
+        getBuilder.addStatement("UdtValue $L = source.getUdtValue($S)", udtValueName, cqlName);
```

Comment:
Types should be injected with `$T` even if they are known statically. Otherwise the generated code might not have the import (in this case I think it works because the set method already imports it).
```suggestion
        getBuilder.addStatement("$T $L = source.getUdtValue($S)", UdtValue.class, udtValueName, cqlName);
```
Also this more refactoring-friendly (although `UdtValue` is unlikely to get renamed at this stage).